rust/src/action/mod.rs

```diff
@@ -791,11 +791,7 @@ pub(crate) async fn find_latest_check_point_for_version(
                 continue;
             }
             if cp.is_none() || curr_ver > cp.unwrap().version {
-                cp = Some(CheckPoint {
-                    version: curr_ver,
-                    size: 0,
-                    parts: None,
-                });
+                cp = Some(CheckPoint::new(version, 0, None));
```

Comment:
@wjones127 this was the culprit. Fixing...

```suggestion
                cp = Some(CheckPoint::new(curr_ver, 0, None));
```